spot-contracts/contracts/strategies/FeeStrategy.sol

```diff
@@ -0,0 +1,227 @@
+        int256 upper;
+        /// @notice Sigmoid slope
+        int256 growth;
+    }
+
+    /// @notice Parameters which control the asymptotes and the slope of the yearly perp token's rollover fee.
+    SigmoidParams public rolloverFeeAPR;
+
+    /// @notice Contract constructor.
+    /// @param perp_ Reference to perp.
+    /// @param vault_ Reference to the vault.
+    constructor(IPerpetualTranche perp_, IVault vault_) {
+        perp = perp_;
+        vault = vault_;
+    }
+
+    /// @notice Contract initializer.
+    function init() public initializer {
+        __Ownable_init();
+        maxMintFeePerc = HUNDRED_PERC / 40; // 2.5%
+        maxBurnFeePerc = HUNDRED_PERC / 40; // 2.5%
+        rolloverFeeAPR.lower = -int256(HUNDRED_PERC) / 50; // -2%
+        rolloverFeeAPR.upper = int256(HUNDRED_PERC) / 20; // 5%
+        rolloverFeeAPR.growth = 5 * int256(HUNDRED_PERC); // 5x
+        deviationThresholdPerc = HUNDRED_PERC / 20; // 5%
+    }
+
+    /// @notice Updates the deviation treshold percentage.
+    /// @param deviationThresholdPerc_ The new deviation threshold percentage
+    ///        as a fixed point number with {PERC_DECIMALS} places.
+    function updateDeviationTreshold(uint256 deviationThresholdPerc_) external onlyOwner {
+        require(deviationThresholdPerc_ <= HUNDRED_PERC, "FeeStrategy: mint fee too high");
+        deviationThresholdPerc = deviationThresholdPerc_;
+    }
+
+    /// @notice Updates the mint fee parameters.
+    /// @param maxMintFeePerc_ The new mint fee ceiling percentage
+    ///        as a fixed point number with {PERC_DECIMALS} places.
+    function updateMintFees(uint256 maxMintFeePerc_) external onlyOwner {
+        require(maxMintFeePerc_ <= HUNDRED_PERC, "FeeStrategy: mint fee too high");
+        maxMintFeePerc = maxMintFeePerc_;
+    }
+
+    /// @notice Updates the burn fee parameters.
+    /// @param maxBurnFeePerc_ The new burn fee ceiling percentage
+    ///        as a fixed point number with {PERC_DECIMALS} places.
+    function updateBurnFees(uint256 maxBurnFeePerc_) external onlyOwner {
+        require(maxBurnFeePerc_ <= HUNDRED_PERC, "FeeStrategy: burn fee too high");
+        maxBurnFeePerc = maxBurnFeePerc_;
+    }
+
+    /// @notice Update the parameters determining the slope and asymptotes of the sigmoid fee curve.
+    /// @param p Lower, Upper and Growth sigmoid paramters are fixed point numbers with {PERC_DECIMALS} places.
+    function updateRolloverFees(SigmoidParams memory p) external onlyOwner {
+        require(p.lower >= -int256(MAX_PERC), "FeeStrategy: fee bound too low");
+        require(p.upper <= int256(MAX_PERC), "FeeStrategy: fee bound too high");
+        require(p.lower < p.upper, "FeeStrategy: paramters invalid");
+        rolloverFeeAPR.lower = p.lower;
+        rolloverFeeAPR.upper = p.upper;
+        rolloverFeeAPR.growth = p.growth;
+    }
+
+    /// @inheritdoc IFeeStrategy
+    function computeMintFeePerc() external override returns (uint256) {
```

Comment:
Could this be a view function?

Reply:
Since perp's `getTVL` calls `updateState` this can't be a view function